### ISSUE TITLE
Mocks incompatible with value classes

### ISSUE THREAD

**Issue by b3er** [lupuuss/Mokkery#2]
It seems that using value class, e.g. `kotlin.Result` in mocks will fail the test with exception

Given the interface
```
interface TestInterfaceWithValueClass {
    fun test(): Result<String>
}
```
Calling the test
```
    @Test
    fun testReturnsMockableInterfaceWithValueClassImpl() {
        val mocked = mock<TestInterfaceWithValueClass> {
            every { test() } returns Result.success("test")
        }
        assertEquals(Result.success("test"), mocked.test())
    }
```
Will fail with exception
```
java.lang.NullPointerException: Cannot invoke "kotlin.Result.unbox-impl()" because the return value of "dev.mokkery.internal.MokkeryInterceptor.interceptCall(dev.mokkery.internal.CallContext)" is null
	at dev.mokkery.test.TestInterfaceWithValueClassb7d4c11efdfd4d2aa2c67574e6dd5d5eMock.test-d1pmJ48(MockTest.kt)
	at dev.mokkery.test.MockTest$testReturnsMockableInterfaceWithValueClassImpl$mocked$1$1.invoke-IoAF18A(MockTest.kt:23)
	at dev.mokkery.test.MockTest$testReturnsMockableInterfaceWithValueClassImpl$mocked$1$1.invoke(MockTest.kt)
	at dev.mokkery.internal.EveryKt.internalEvery(Every.kt:21)
	at dev.mokkery.test.MockTest$testReturnsMockableInterfaceWithValueClassImpl$mocked$1.invoke(MockTest.kt)
	at dev.mokkery.test.MockTest$testReturnsMockableInterfaceWithValueClassImpl$mocked$1.invoke(MockTest.kt:22)
	at dev.mokkery.test.TestInterfaceWithValueClassb7d4c11efdfd4d2aa2c67574e6dd5d5eMock.<init>(MockTest.kt)
	at dev.mokkery.test.MockTest.testReturnsMockableInterfaceWithValueClassImpl(MockTest.kt:22)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	...
```
mokkery version: 1.9.23-1.6.0
kotlin version: 1.9.23

**Comment by lupuuss** [comment]
I was able to reproduce it. It affects JVM and Wasm. Fixing it for JVM should not be a big problem, but it might be problematic for  Wasm. Anyway, I will do my best to provide a fix.

**Comment by b3er** [comment]
Thanks! Fixing for jvm for me is the only needed (atm, haha). The library looks nice, but lack of this feature prevents for adoption.

**Comment by lupuuss** [comment]
Fixed in Mokkery 1.6.1. Thank you for reporting the issue!
* For JVM all value classes should work properly now.
* For Wasm it fixes only standard library value classes (`Result`, `Duration`). Other value classes have to be registered with:
```kotlin
AutofillProvider.forInternals.types.register { ValueClass(null) }
````
* Other targets were not affected.
